### PR TITLE
[SPARK-8483][Streaming] Remove commons-lang3 dependency from Flume Si…

### PR DESCRIPTION
**Pull request by harishreedharan**
…nk. Also bump Flume version to 1.6.0

### CODE REVIEW

File: external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ private[flume] class SparkAvroCallbackHandler(val threads: Int, val channel: Cha
   // Since the new txn may not have the same sequence number we must guard against accidentally
   // committing a new transaction. To reduce the probability of that happening a random string is
   // prepended to the sequence number. Does not change for life of sink
-  private val seqBase = RandomStringUtils.randomAlphanumeric(8)
+  private val seqBase = UUID.randomUUID().toString.substring(0, 8)
```
Comment (JoshRosen) [inline external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala]: Will this be sufficiently random?

Comment (JoshRosen) [inline external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala]: Why not just use the Scala random string functionality instead?

Comment (harishreedharan) [inline external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala]: Last time, we tried it and we ended up getting the same string multiple times. I am not entirely sure why, but that was the reason we used RandomStringUtils in the first place.

The idea of using the seqBase is to protect against the case where a sink restart can cause collisions. I think this should be random enough, else we can use the full UUID which I feel is adding too much overhead without a whole lot of gain.

**Comment by tdas** [comment]
Could you please make a separate JIRA and PR for bumping the version. These are two completely different issues.

File: external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala (hunk 1 above)
Comment (JoshRosen) [inline external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala]: Ah, this is a Type 4 random UUID so it only relies on random numbers and not something less random, such as the date / time or a MAC address or namespace, so this seems fine to me.

**Comment by harishreedharan** [comment]
Looks like github plugin fail

**Comment by harishreedharan** [comment]
Jenkins, test this please

**Comment by harishreedharan** [comment]
Jenkins, test this please

**Comment by andrewor14** [comment]
@JoshRosen seems like this is not running pyspark tests. Do you know what the issue is?

**Comment by harishreedharan** [comment]
It took only about 15 mins, so it seems to have run only a few tests or none at all.

**Comment by andrewor14** [comment]
ah, looks like there are no flume python tests. Since this only touches flume code it doesn't have to run streaming tests.

**Comment by tdas** [comment]
dude, you didnt update the title and PR message. Its going to be super confusing for people later. Please update asap, and I can include it in 1.4.1

**Comment by tdas** [comment]
Merged to master and 1.4. But I think I missed the RC1 cutoff.

**Comment by harishreedharan** [comment]
Oops, missed the title. Looks like this was merged though - there is another PR for the bump to 1.6.0

**Pull request closed without merging**